joomla/joomla-cms
# [5.0] Set created_by and modified_by user for guided tours and tour steps on new installation

Pull Request for Issue # .

### Summary of Changes

Currently new guided tours and guided tour steps are inserted on new installations with created_by and modified_by user IDs = zero in files "installation/sql/`<dbtype>`/extensions.sql".

This results in these user IDs being zero until the tours or steps are edited for the first time because the conversion to the real user ID happens only when that ID is greater than zero and when the database tables and columns are in the list to be updated here: https://github.com/joomla/joomla-cms/blob/5.0-dev/installation/src/Model/ConfigurationModel.php#L273-L303

This PR here fixes that.

### Testing Instructions

Make a new installation and then check the `created_by` and `modified_by` columns in the `#__guidedtours` and `#__guidedtour_steps` tables in database e.g. with phpMyAdmin.

### Actual result BEFORE applying this Pull Request

The `created_by` and `modified_by` columns in the `#__guidedtours` and `#__guidedtour_steps` tables have value zero after a new installation.

### Expected result AFTER applying this Pull Request

The `created_by` and `modified_by` columns in the `#__guidedtours` and `#__guidedtour_steps` tables are equal to the super user's user ID after a new installation.

### Link to documentations
Please select:
